arch/x86/apic.c

```diff
@@ -179,6 +178,9 @@ void init_apic_timer(void) {
     apic_lvt_timer_t timer;
     printk("Initializing local APIC timer\n");
 
+    /* Enable interrupts for calibration */
+    sti();
```

Comment:
Looks like interrupts are already enabled earlier in kernel_start(), so should this be a flags saving enable instead? `"pushf; sti;"` ....` "popf"`?

Reply:
Yes, although not necessary at the moment, this is good idea to add proper API to do such intr enable/disable and use it instead of the current one leaving side-effects behind. Will do.